docs/docs/cloud/how-tos/stream_messages.md

```diff
@@ -0,0 +1,391 @@
+# How to stream messages from your graph
+
+LangGraph Cloud supports multiple streaming modes. The main ones are:
+
+- `values`: This streaming mode streams back values of the graph. This is the **full state of the graph** after each node is called.
+- `updates`: This streaming mode streams back updates to the graph. This is the **update to the state of the graph** after each node is called.
+- `messages`: This streaming mode streams back messages - both complete messages (at the end of a node) as well as **tokens** for any messages generated inside a node. This mode is primarily meant for powering chat applications.
+
+
+This guide covers `stream_mode="messages"`.
+
+In order to use this mode, the state of the graph you are interacting with MUST have a messages key that is a list of messages.
+Eg, the state should look something like the following code block. Note that LangGraph Cloud only supports hosting graphs written in Python at the moment,
+but JS support is on the way.
+
+
+    ```python
+    from typing import TypedDict, Annotated
+    from langgraph.graph import add_messages
+    from langchain_core.messages import AnyMessage
+
+    class State(TypedDict):
+        messages: Annotated[list[AnyMessage], add_messages]
+    ```
+
+
+OR it should be an instance or subclass of `from langgraph.graph import MessageState` (`MessageState` is just a helper type hint equivalent to the above).
+
+With `stream_mode="messages"` two things will be streamed back:
+
+- It outputs messages produced by any chat model called inside (unless tagged in a special way)
+- It outputs messages returned from nodes (to allow for nodes to return `ToolMessages` and the like
+
+First let's set up our client and thread:
+
+=== "Python"
+    
+    ```python
+    from langgraph_sdk import get_client
+
+    client = get_client(url="whatever-your-deployment-url-is")
+    # create thread
+    thread = await client.threads.create()
+    print(thread)
+    ```
+
+=== "Javascript"
+
+    ```js
+    import { Client } from "@langchain/langgraph-sdk";
+
+    const client = new Client({apiUrl:"whatever-your-deployment-url-is"});
+    # create thread
+    const thread = await client.threads.create();
+    console.log(thread)
+    ```
+
+
+Output:
+
+
+
+    {'thread_id': 'e1431c95-e241-4d1d-a252-27eceb1e5c86',
+     'created_at': '2024-06-21T15:48:59.808924+00:00',
+     'updated_at': '2024-06-21T15:48:59.808924+00:00',
+     'metadata': {}}
+
+First let's write a quick helper function:
```

Comment:
we already say `First` above, let's just say "We'll use a helper function for better rendering of the tool calls"